packages/databricks-vscode/src/extension.ts

```diff
@@ -574,14 +578,51 @@ export async function activate(
 
     CustomWhenContext.setActivated(true);
     telemetry.recordEvent(Events.EXTENSION_ACTIVATED);
-
+    context.subscriptions.push(
+        debug.registerDebugConfigurationProvider(
+            "python",
+            new DatabricksDebugConfigProvider(context)
+        )
+    );
     const publicApi: PublicApi = {
         version: 1,
         connectionManager: connectionManager,
     };
     return publicApi;
 }
 
+class DatabricksDebugConfigProvider implements DebugConfigurationProvider {
+    constructor(private readonly context: ExtensionContext) {}
+
+    // async provideDebugConfigurations() {
+    //     return [
+    //         {
+    //             type: "databricks",
+    //             name: `Databricks: File`,
+    //             program: "${file}",
+    //             request: "launch",
+    //         },
+    //     ];
+    // }
+
+    resolveDebugConfigurationWithSubstitutedVariables(
+        folder: WorkspaceFolder | undefined,
+        debugConfiguration: DebugConfiguration
+    ): ProviderResult<DebugConfiguration> {
+        if (debugConfiguration.type !== "python") {
```

Comment:
[VSCode makes sure](https://github.com/microsoft/vscode/blob/87fd6d3320d4ce70ba896bbf273b960a3f7c200f/src/vs/workbench/contrib/debug/browser/debugConfigurationManager.ts#L150) that you won't get a config for a type you haven't registered this config provider, so this check is not required